hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java

```diff
@@ -893,6 +906,9 @@ public short getDefaultReplication(Path f) {
       return res.targetFileSystem.getDefaultReplication(res.remainingPath);
     } catch (FileNotFoundException e) {
       throw new NotInMountpointException(f, "getDefaultReplication"); 
+    } catch (IOException e) {
+      throw new RuntimeException("Not able to initialize fs in "
```

Comment:
FYI. I've got a WrappedIOException, but in #2069 making it a public `org.apache.hadoop.fs.functional.RuntimeIOException` whose cause is only ever IOE. Not something to be picked up here (yet), but worth knowing. I'm trying to make the functional & lambda/expression stuff more useable,